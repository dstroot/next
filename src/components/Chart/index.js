import React from 'react';
import PropTypes from 'prop-types';

// recharts
import {
  AreaChart,
  Area,
  // XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  ResponsiveContainer,
} from 'recharts';

// currency formatter
const formatter = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
});

const Chart = ({ data }) => (
  <>
    <ResponsiveContainer height={400}>
      <AreaChart data={data} margin={{ top: 0, right: 0, left: 45, bottom: 0 }}>
        <CartesianGrid strokeDasharray="3 3" />
        {/* <XAxis /> */}
        <YAxis tickFormatter={value => formatter.format(value).slice(0, -3)} />
        <Tooltip formatter={value => formatter.format(value).slice(0, -3)} />
        <Area
          type="monotone"
          dataKey="savings"
          stackId="1"
          stroke="#CCE1F1"
          fill="#CCE1F1"
        />
        <Area
          type="monotone"
          dataKey="investment"
          stackId="1"
          stroke="#3F92C8"
          fill="#3F92C8"
        />
        <Area
          type="monotone"
          dataKey="annuity"
          stackId="1"
          stroke="#007DBD"
          fill="#007DBD"
        />
      </AreaChart>
    </ResponsiveContainer>
    <div className="d-flex justify-content-between mb-3 text-muted">
      <div className="ml-5">Today</div>
      <div>30 Years</div>
    </div>
  </>
);

Chart.propTypes = {
  data: PropTypes.array.isRequired,
};

export default Chart;
