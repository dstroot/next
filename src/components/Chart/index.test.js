import React from 'react';
import { create } from 'react-test-renderer';

import Chart from '../Chart';

let data = [
  {
    year: 1,
    savings: 63573.31,
    investment: 686.2300000000032,
    annuity: 129.47999999999593,
  },
  {
    year: 2,
    savings: 77531.59,
    investment: 1568.449999999997,
    annuity: 297.84000000001106,
  },
  {
    year: 3,
    savings: 91885.76,
    investment: 2659.4000000000087,
    annuity: 508.1299999999901,
  },
  {
    year: 4,
    savings: 106647.04,
    investment: 3972.4900000000052,
    annuity: 763.570000000007,
  },
  {
    year: 5,
    savings: 121826.98,
    investment: 5521.830000000002,
    annuity: 1067.5599999999977,
  },
  {
    year: 6,
    savings: 137437.46,
    investment: 7322.200000000012,
    annuity: 1423.7300000000105,
  },
  {
    year: 7,
    savings: 153490.68,
    investment: 9389.170000000013,
    annuity: 1835.929999999993,
  },
  {
    year: 8,
    savings: 169999.21,
    investment: 11739.080000000016,
    annuity: 2308.209999999992,
  },
  {
    year: 9,
    savings: 186975.96,
    investment: 14389.089999999997,
    annuity: 2844.9000000000233,
  },
];

function createNodeMock() {
  const doc = document.implementation.createHTMLDocument();
  return { parentElement: doc.body };
}

describe('chart', () => {
  it('it should render', () => {
    let tree = create(<Chart data={data} />, { createNodeMock });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
