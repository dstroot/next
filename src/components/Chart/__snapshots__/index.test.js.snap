// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chart it should render 1`] = `
Array [
  <div
    className="recharts-responsive-container"
    style={
      Object {
        "height": 400,
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "display": "none",
          "height": 0,
          "position": "absolute",
          "visibility": "hidden",
          "width": 0,
        }
      }
    />
  </div>,
  <div
    className="d-flex justify-content-between mb-3 text-muted"
  >
    <div
      className="p-2 font-weight-bold ml-5"
    >
      Today
    </div>
    <div
      className="p-2 font-weight-bold"
    >
      30 Years
    </div>
  </div>,
]
`;
